blockchain/src/blockchain/history_sync.rs

```diff
@@ -307,28 +149,149 @@ impl Blockchain {
             // Check if the receipts contain an error.
             if let Err(e) = receipts {
                 warn!(
-                    %block,
-                    reason = "commit of block failed",
+                    %self.block,
+                    reason = "commit of macro block failed",
                     block_no = block_numbers[i],
-                    num_transactions = block_transactions[i].len(),
                     num_inherents = block_inherents[i].len(),
                     error = &e as &dyn Error,
                     "Rejecting block",
                 );
 
                 txn.abort();
                 #[cfg(feature = "metrics")]
-                this.metrics.note_invalid_block();
+                blockchain.metrics.note_invalid_block();
                 return Err(PushError::AccountsError(e));
             }
         }
-        this.state.accounts.finalize_batch(&mut txn);
+        blockchain.state.accounts.finalize_batch(&mut txn);
```

Comment:
I think this is a confusing name, because we are not finalizing a batch.
"update root" or something similar would be a better name

Reply:
Mmm, this is an accounts batch. I don't have the full context but there is a family of functions in the accounts tree talking about batches but they doesn't look related to the concept of a block batch